### PR TITLE
HOST GPT-2

### PR DESCRIPTION
**Pull request by IonesioJunior**
Turn Grid Nodes/Lib be able to host huge models
- ADD new notebook [Host GPT-2 on GridNode](https://github.com/OpenMined/Grid/blob/hosting_huge_models/examples/Serving%20models/Host%20GPT-2%20(12-layer%2C%20768-hidden%2C%2012-heads%2C%20117M%20parameters)%20.ipynb)

### CODE REVIEW

File: grid/client.py [hunk 1]
```diff
@@ -57,6 +60,20 @@ def _send_http_request(
 
         return response
 
+    def _send_streaming_post(
+        self, route, data=None, N: int = 10, unhexlify: bool = True
```
Comment (mari-linhares) [inline grid/client.py]: unhexlify and N argument are not needed

File: grid/client.py [hunk 2]
```diff
@@ -57,6 +60,20 @@ def _send_http_request(
...
+    ):
```
Comment (mari-linhares) [inline grid/client.py]: Can you add a docstring explaining why this is needed / what this method implements?

File: app/websocket/app/main/routes.py [hunk 3]
```diff
@@ -53,7 +53,11 @@ def model_inference(model_id):
     # to the local worker in order to execute it
     sy.hook.local_worker.register_obj(data)
 
-    response = model(data).detach().numpy().tolist()
+    response = model(data)
```
Comment (mari-linhares) [inline app/websocket/app/main/routes.py]: add a comment describing why this check is needed.

File: app/websocket/app/main/routes.py [hunk 4]
```diff
@@ -66,9 +70,15 @@ def model_inference(model_id):
 @main.route("/serve-model/", methods=["POST"])
 def serve_model():
     encoding = request.form["encoding"]
-    serialized_model = request.form["model"].encode(encoding)
     model_id = request.form["model_id"]
 
+    if request.files:
```
Comment (mari-linhares) [inline app/websocket/app/main/routes.py]: add a comment describing why this check is needed.

File: examples/Serving models/Host GPT-2 (12-layer, 768-hidden, 12-heads, 117M parameters) .ipynb [hunk 5]
```diff
@@ -0,0 +1,178 @@
+{
```
Comment (mari-linhares) [inline examples/Serving models/Host GPT-2 (12-layer, 768-hidden, 12-heads, 117M parameters) .ipynb]: Can you add instructions about how to install pytorch_transformers and details about the GPT-2 version being used?

File: examples/Serving models/Host GPT-2 (12-layer, 768-hidden, 12-heads, 117M parameters) .ipynb [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Predicted text:   Who was Jim Henson? Jim Henson was aage\n",
+      "CPU times: user 26.5 ms, sys: 0 ns, total: 26.5 ms\n",
+      "Wall time: 174 ms\n"
+     ]
+    }
+   ],
+   "source": [
```
Comment (mari-linhares) [inline examples/Serving models/Host GPT-2 (12-layer, 768-hidden, 12-heads, 117M parameters) .ipynb]: We should try something like this: https://github.com/huggingface/pytorch-transformers/blob/master/examples/run_generation.py#L106 :eyes:

File: examples/Serving models/Host GPT-2 (12-layer, 768-hidden, 12-heads, 117M parameters) .ipynb (hunk 5 above)
Comment (IonesioJunior) [reply examples/Serving models/Host GPT-2 (12-layer, 768-hidden, 12-heads, 117M parameters) .ipynb]: Done! (b017e19)

File: app/websocket/app/main/routes.py (hunk 4 above)
Comment (IonesioJunior) [reply app/websocket/app/main/routes.py]: Done! (b017e19)

File: app/websocket/app/main/routes.py (hunk 3 above)
Comment (IonesioJunior) [reply app/websocket/app/main/routes.py]: Done! (b017e19)

File: grid/client.py (hunk 2 above)
Comment (IonesioJunior) [reply grid/client.py]: Done! (b017e19)

File: grid/client.py (hunk 1 above)
Comment (IonesioJunior) [reply grid/client.py]: Done! (b017e19)

File: examples/Serving models/Host GPT-2 (12-layer, 768-hidden, 12-heads, 117M parameters) .ipynb (hunk 6 above)
Comment (IonesioJunior) [reply examples/Serving models/Host GPT-2 (12-layer, 768-hidden, 12-heads, 117M parameters) .ipynb]: Done! (d491e0a)

**Pull request merged**